### PR TITLE
DEV-37: Use eslint to lint custom javascript

### PR DESCRIPTION
**Pull request by jesconstantine**
User story: [DEV-37: Use Drupal core JS and CSS linting as default](https://palantir.atlassian.net/browse/DEV-37)

## Description

This PR adds the necessary configuration and scripting to use eslint on our custom js in projects.

## Testing instructions

1. Run `composer create-project palantirnet/drupal-skeleton test dev-dev-37-eslint`
1. Follow the [updated readme "Quick Start" steps #2-3](https://github.com/palantirnet/drupal-skeleton/pull/146/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5) to get Drupal installed in your test project: https://github.com/palantirnet/drupal-skeleton/pull/146/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5
1. Exit your ddev ssh session 
1. Run `ddev phing code-review` from your host machine
    - [x] Observe that we check for npm dependencies being installed at 2 levels: project and in core.
    - [x] Observe that we install the dependencies.
    - [x] Observe that the code-review passes
1. Add some bad js to `${drupal.root}/modules/custom/bad-js/bad.es6.js`:
    ```const x = 3;
		 const squared = x * x;
		 const cubed = x * x * x;
		 console.log('squared =', squared);
		 console.log('cubed =', cubed);
 
		 const squared2 = Math.pow(x, 2);
		 const cubed2 = Math.pow(x, 3);
		 console.log('squared2 =', squared2);
		 console.log('cubed2 =', cubed2);
 
		 const complex = Math.pow(x * 2, 3 + 4);
		 console.log('complex =', complex);
	```
1. Run `ddev phing code-review` from your host machine
    - [x] Observe that we check for npm dependencies being installed at 2 levels: project and in core.
    - [x] Observe that we no longer installed the dependencies because we found them already.
    - [x] Observe that the PHP linters still pass
    - [x] Observe that eslint finds errors
    - [x] Observe the step fails as it should.

## Notes

This PR relies on https://github.com/palantirnet/the-build/pull/223

### CODE REVIEW

File: README.md
Comment (jesconstantine) [inline README.md]: I actually think we don't need to declare node as a dependency because we install and run it on ddev.  But do we need npm? Does that get installed with node automatically?

File: .nvmrc
Comment (jesconstantine) [inline .nvmrc]: From #144 - thanks Byron!

File: .circleci/config.yml
Comment (jesconstantine) [inline .circleci/config.yml]: From #144 - thanks Byron!

File: README.md
Comment (jesconstantine) [inline README.md:40]: These changes are all just formatting to allow us to just use `1.`

File: README.md
Comment (jesconstantine) [inline README.md]: This is a new addition.

File: composer.json
Comment (jesconstantine) [inline composer.json:19]: This is temporary.  After this PR is approved we should be able to remove.

File: composer.json
Comment (jesconstantine) [inline composer.json:40]: This is temporary.  After this PR is approved we should be able to revert or update.

File: .ddev/commands/web/npx
Comment (jesconstantine) [inline .ddev/commands/web/npx]: Allows us to run `ddev npx ...` Used in the phing task.

File: .ddev/commands/web/phing
Comment (jesconstantine) [inline .ddev/commands/web/phing:5]: Allows us to run `ddev phing ...`

File: .eslintrc.json
Comment (jesconstantine) [inline .eslintrc.json:3]: We just use Drupal Core's rules and config for eslint.

File: README.md
Comment (jesconstantine) [reply README.md:136]: @todo: add a link to name rules here: https://docs.npmjs.com/cli/v8/configuring-npm/package-json#name

File: README.md
Comment (jesconstantine) [reply README.md]: Removing npm as dependency because npm and yarn come on ddev out of the box

**Review by patrickfweston — APPROVED** [review]
Everything here works great for me! I think this will help us out a lot on DHS, so I'd be in favor of merging this in. I'm going to hold off until others have a chance to look though.

**Comment by patrickfweston** [comment]
I do see there is a test failure, but I'm wondering if we can test that in a different way. I'm not sure how relevant the existing test is.

**Pull request merged**
